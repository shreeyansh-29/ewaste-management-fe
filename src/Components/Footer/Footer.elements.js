import styled from "styled-components";

export const FooterContainer = styled.div`
  background-color: #101522;
  padding: 4rem 0 2rem 0;
  display: flex;
  flex-direction: column;
  justify-content: center;
  align-items: center;

  @media screen and (max-width: 768px) {
    padding: 5rem 0 2rem 0;
  }
`;

export const FooterSubscription = styled.section`
  display: flex;
  flex-direction: column;
  justify-content: center;
  align-items: center;
  text-align: center;
  margin-bottom: 24px;
  padding: 24px;
  color: #fff;
`;

export const FooterSubHeading = styled.p`
  font-family: "Trebuchet MS", "Lucida Sans Unicode", "Lucida Grande",
    "Lucida Sans", Arial, sans-serif;
  margin-bottom: 24px;
  font-size: 24px;
  @media screen and (max-width: 820px) {
    font-size: 20px;
  }
`;

export const FooterSubText = styled.p`
  margin-bottom: 24px;
  font-size: 20px;
`;

export const Form = styled.form`
  display: flex;
  justify-content: center;
  align-items: center;
  @media screen and (max-width: 820px) {
    flex-direction: column;
    width: 80%;
  }
`;

export const SocialMedia = styled.section`
  max-width: 1000px;
  width: 100%;
`;

export const SocialMediaWrap = styled.div`
  display: flex;
  justify-content: space-between;
  align-items: center;
  width: 90%;
  max-width: 1000px;
  margin: 40px auto 0 auto;
  @media screen and (max-width: 820px) {
    flex-direction: column;
  }
`;

export const WebsiteRights = styled.small`
  color: #fff;
  margin-left: auto;
  margin-right: auto;
  font-size: 15px;
`;

export const SocialIcons = styled.div`
  display: flex;
  justify-content: space-between;
  align-items: center;
  width: 240px;
`;

export const SocialIconLink = styled.a`
  color: #fff;
  font-size: 24px;
`;
