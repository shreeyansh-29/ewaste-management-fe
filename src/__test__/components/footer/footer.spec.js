/* eslint-disable no-undef */
import React from "react";
import Footer from "../../../components/footer/footer";
import Enzyme, {shallow} from "enzyme";
import toJson from "enzyme-to-json";
import Adapter from "@wojtekmaj/enzyme-adapter-react-17";

Enzyme.configure({adapter: new Adapter()});

const mockedUsedNavigate = jest.fn();
jest.mock("react-router-dom", () => ({
  ...jest.requireActual("react-router-dom"),
  useNavigate: () => mockedUsedNavigate,
}));

it("test footer", () => {
  const handleClick = jest.fn();
  expect.assertions(1);
  expect(
    toJson(shallow(<Footer handleClick={handleClick} />))
  ).toMatchSnapshot();
});
