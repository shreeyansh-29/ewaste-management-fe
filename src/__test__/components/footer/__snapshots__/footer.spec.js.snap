// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test footer 1`] = `
<Fragment>
  <styled.div>
    <styled.section>
      <styled.p>
        Join our exclusive platform
      </styled.p>
      <styled.p>
        You can join at any time.
      </styled.p>
      <styled.form>
        <styled.button
          fontBig={true}
          onClick={[Function]}
        >
          Join
        </styled.button>
      </styled.form>
    </styled.section>
    <styled.section>
      <styled.div>
        <styled.small>
          â’¸ 
          2022
           Powered by E-Waste Management
        </styled.small>
      </styled.div>
    </styled.section>
  </styled.div>
</Fragment>
`;
